### PR TITLE
Fix env folder

### PR DESCRIPTION
**Pull request by SparkSnail**
1. Move trials folder under env/{envId} folder to map trial jobs
2. Remove remote mode dup env folder

### CODE REVIEW

File: nni/tools/trial_tool/trial.py [hunk 1]
```diff
@@ -47,7 +47,7 @@ def run(self):
 
         nni_log(LogType.Info, "%s: start to run trial" % self.name)
 
-        trial_working_dir = os.path.realpath(os.path.join(os.curdir, "..", "..", "trials", self.id))
+        trial_working_dir = os.path.realpath(os.path.join(os.curdir, "trials", self.id))
```
Comment (J-shang) [inline nni/tools/trial_tool/trial.py:50]: There is a concern that if users want to find their outputs, they need know the trail's runner. And do we have a place to save these mapping?

Comment (SparkSnail) [reply nni/tools/trial_tool/trial.py:50]: Yes, already add trials' envId and platform information in `nnictl trial ls` and webui.

Comment (J-shang) [reply nni/tools/trial_tool/trial.py:50]: yes, I find `envId`, but it seems there no place to store the `runnerId`, that I need it to splice path like `expId/envs/runnerId/trials/trialId`. And I am worried if the checkpoint or output is easy to find under this way. Before, I know it is under `expId/trials/trialId`, but now I need a way to get `runnerId`.

Comment (J-shang) [reply nni/tools/trial_tool/trial.py:50]: Oh, I find we saved `envName`, but no `envId`, I made a mistake upstairs.

**Review by J-shang — APPROVED** [review]

**Review by liuzhe-lz — APPROVED** [review]

**Pull request merged**